### PR TITLE
Feat(eos_designs): Add support for Zscaler internet-exit policy

### PR DESCRIPTION
**Pull request by gmuloc**
## Change Summary

Add support for Zscaler internet-exit policy

## Requires

- [x] #3748 
- [x] #3752 
- [x] #3824
- [x] #3825 
- [x] #3832 
- [x] #3867
- [x] #3875

## Component(s) name

`arista.avd.eos_desigms`

## Proposed changes

- [x] Add schema
  - [x] AVT `internet_exit.policy`
  - [x] `cv_pathfinder_internet_exit_policies`
  - [x] `zscaler_endpoints`
  - [x] WAN Interfaces `internet_exit.policies`
- [x] Add `eos_designs` code
  - [x] Internal data parsing
  - [x] `monitor_connectivity`
  - [x] `router_service_insertion`
  - [x] `router_internet_exit`
  - [x] `tunnel_interfaces`
      - [x] NAT profile on interface
  - [x] `ethernet_interfaces` (already there)
  - [x] `router_adaptive_virtual_topology`
  - [x] `ip_nat`
  - [x] `ip_access_lists`
  - [x] `ip_security`
      - [x] Type 7 obfuscation 
- [x] Add option for encryption or not on tunnels.
- [ ] Add tests
    - [x] Negative unit tests for clashing tunnels
    - [ ] ~Negative unit tests for clashing monitor connectivity - not triggered once clashing tunnels is caught~
- [ ] Add doc

## Future work

tracked in https://github.com/aristanetworks/avd-internal/issues/156

- [ ] may require a host entry for the http://gateway.zscalerbeta.net/vpntest
- [ ] Add metadata
  - [ ] New model for Zscaler metadata in the metadata studio
  - [ ] Structured config model/schema for required metadata
  - [ ] Code to generate new metadata inputs in `cv_workflow`
  - [ ] Code to generate new structured config metadata in `eos_designs`
- [ ] Revisit/review naming, descriptions of all generated config objects.

## How to test

molecule

### Questions

- [x] Do we need to have the internet-exit policies on the CV-Pathfinder or not? (think not but..)
    No for now - will change it later if required.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg [hunk 1]
```diff
@@ -317,6 +413,56 @@ application traffic recognition
...
+   interface set ZSTunnel102 Tunnel102
+   interface set ZSTunnel110 Tunnel110
+   interface set ZSTunnel111 Tunnel111
+   interface set ZSTunnel112 Tunnel112
+   !
+   host ZSTunnel100
+      description
+      Zscaler tunnel
+      local-interfaces ZSTunnel100 address-only
+      ip 172.25.42.16/31
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: ip without mask that should be on the other side of the tunnel

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg [hunk 2]
```diff
@@ -317,6 +413,56 @@ application traffic recognition
...
+   interface set ZSTunnel100 Tunnel100
+   interface set ZSTunnel101 Tunnel101
+   interface set ZSTunnel102 Tunnel102
+   interface set ZSTunnel110 Tunnel110
+   interface set ZSTunnel111 Tunnel111
+   interface set ZSTunnel112 Tunnel112
+   !
+   host ZSTunnel100
+      description
+      Zscaler tunnel
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: ???

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg [hunk 3]
```diff
@@ -317,6 +413,56 @@ application traffic recognition
...
+      local-interfaces ZSTunnel101 address-only
+      ip 172.25.42.16/31
+      url http://TODO
+   !
+   host ZSTunnel102
+      description
+      Zscaler tunnel
+      local-interfaces ZSTunnel102 address-only
+      ip 172.25.42.16/31
+      url http://TODO
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: may require a host entry for the `http://gateway.zscalerbeta.net/vpntest`

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg [hunk 4]
```diff
@@ -101,6 +103,32 @@ router adaptive-virtual-topology
...
+        local connection ZSCALER-CONN-110
+    !
+    exit-group ZSCALER-EXIT-POLICY-2_SEC
+        local connection ZSCALER-CONN-111
+    !
+    exit-group ZSCALER-EXIT-POLICY-2_TER
+        local connection ZSCALER-CONN-112
+    !
+    policy ZSCALER-EXIT-POLICY-1
+        exit-group system-default-exit-group
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: missing exit groups

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/zscaler_endpoints.schema.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+          city:
+            type: str
+            required: true
+          country:
+            type: str
+            required: true
+          latitude:
+            type: str
+            convert_types:
+              - int
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/zscaler_endpoints.schema.yml:50]: Add float for longitude and latitude

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge1.cfg]: We can just add a limitation that you must have name server configured.

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-pathfinder.cfg [hunk 6]
```diff
@@ -87,9 +87,11 @@ router adaptive-virtual-topology
       path-selection load-balance LB-PROD-AVT-POLICY-MPLS-ONLY
    !
    profile PROD-AVT-POLICY-VIDEO
+      internet-exit policy ZSCALER-EXIT-POLICY-2
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-pathfinder.cfg]: this should not be there - need to fix it

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml [hunk 7]
```diff
@@ -117,6 +122,13 @@ wan_router:
               wan_circuit_id: 404
               dhcp_accept_default_route: true
               ip_address: dhcp
+              peer_ip: 172.25.42.16/31
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml]: not needed

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-pathfinder.cfg (hunk 6 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-pathfinder.cfg]: fixed

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml [hunk 8]
```diff
@@ -530,3 +545,42 @@ application_classification:
...
+      ipsec_key_salt: THIS_SHOULD_BE_VAULTED
+  - name: ZSCALER-EXIT-POLICY-2
+    fallback_to_system_default: False
+    type: zscaler
+    zscaler:
+      cloud_name: zscalerbeta
+      domain_name: test.local
+      ipsec_key_salt: THIS_SHOULD_BE_VAULTED
+
+# TODO move to its own test as planned
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md [hunk 9]
```diff
@@ -798,6 +798,16 @@ roles/eos_designs/docs/tables/wan-virtual-topologies.md
...
 --8<--
 
+#### Internet Exit policies
+
+!!! note
+
+    This section is only relevant for CV Pathfinder and not for AutoVPN
+
+--8<--
+roles/eos_designs/docs/tables/cv-pathfinder-internet-exit-policies.md
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md:808]: need to add Zscaler table

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md:808]: they are already in the table

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge.cfg [hunk 10]
```diff
@@ -108,6 +113,23 @@ router adaptive-virtual-topology
       avt profile PROD-AVT-POLICY-VIDEO id 4
       avt profile PROD-AVT-POLICY-MPLS-ONLY id 5
 !
+router internet-exit
+    !
+    exit-group ZSCALER-EXIT-POLICY-1_PRI
+        local connection ZSCALER-CONN-100
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge.cfg]: and adjust connection names
```suggestion
        local connection IE-Tunnel100
```

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge.cfg [hunk 11]
```diff
@@ -339,6 +420,35 @@ application traffic recognition
...
+      url http://gateway.zscalerbeta.net/vpntest
+   !
+   host IE-Tunnel102
+      description
+      Internet Exit ZSCALER-EXIT-POLICY-1 TER
+      local-interfaces SET-Tunnel102
+      ip 10.50.9.1
+      url http://gateway.zscalerbeta.net/vpntest
+!
+ip access-list ALLOW_ALL
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/intended/configs/cv-pathfinder-edge.cfg]: check dash / underscore for these objects

File: ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml [hunk 12]
```diff
@@ -97,6 +97,11 @@ wan_router:
               wan_circuit_id: AF830
               ip_address: 172.20.20.20/31
               connected_to_pathfinder: false
+              peer_ip: 172.20.20.21/31
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_designs_unit_tests/inventory/group_vars/CV_PATHFINDER_TESTS.yml]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 13]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+
+- Local exit toward the ISP
+- Local exit toward an enterprise firewall
+- Local exit toward a Secure Internet Gateway
+
+#### AVD abstraction
+
+The internet-exit policies are defined as global variables for all the WAN routers under `cv_pathfinder_internet_exit_policies`.
+
+- A device would be configured with an Internet-exit policy if the internet-exit policy is configured under one of its WAN interfaces and on a policy applied on the device.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]:
```suggestion
- A device would be configured with an Internet-exit policy if the internet-exit policy is configured under one of its WAN interfaces and on a WAN virtual topology policy applied on the device.
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 14]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+          wan_circuit_id: 404
+          dhcp_accept_default_route: true
+          ip_address: dhcp
+          cv_pathfinder_internet_exit: # (1)!
+            policies:
+              - name: ZSCALER-EXIT-POLICY-1
+                tunnel_interface_numbers: 100-102
+              - name: ZSCALER-EXIT-POLICY-2
+                tunnel_interface_numbers: 110-112
+```
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md:527]: Under hieraarchy need to add city and country keys for Zscaler

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 15]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+The policies are assigned a type, currently only `zscaler` is supported. Then additional parameters can be provided according to the type.
+
+```yaml
+cv_pathfinder_internet_exit_policies:
+  - name: ZSCALER-EXIT-POLICY-1
+    type: zscaler
+    zscaler:
+      cloud_name: zscalerbeta
+      domain_name: test.local
+      ipsec_key_salt: THIS_SHOULD_BE_VAULTED
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]:
```suggestion
    # [...] other type specific options
```

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 16]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+!!! Warning "Not currently supported in eos_designs."
+
+#### Local exit toward a Secure Internet Gateway
+
+!!! Warning "Only supported in PREVIEW toward Zscaler SIG"
+
+An internet-exit policy of type `zscaler` leverages the following AVD data model to generate the target configuration.
+The target is for this data to be retrieved from Cloudvision through a lookup plugin for each device to determine what are the best tunnel(s) to use for a given location.
+
+!!! TODO
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: do this

Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: show lookup plugin syntax instead of the retrieved data

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 17]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+    city: Frankfurt
+    datacenter: FRA4
+    country: Germany
+    latitude: 50
+    longitude: 9
+    ip_address: 10.50.9.1
+```
+
+AVD supports up to three tunnels (primary, secondary, tertiary).
+For each `zscaler` type Internet-policies, AVD uses the `cv_pathinfder_internet_exit_policies[name=<POLICY-NAME>].zscaler` dictionary and the `zscaler_endpoints` in combination with the `l3_interfaces.cv_pathfinder_internet_exit.policies[name=<POLICY-NAME>].tunnel_interface_numbers` to generate the internet-exit configuration.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md:558]: Add policy options

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md (hunk 16 above)
Comment (gmuloc) [reply ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: show the cv_token and cv_server

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 18]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+    longitude: -77
+    ip_address: 10.39.77.1
+  tertiary:
+    city: Frankfurt
+    datacenter: FRA4
+    country: Germany
+    latitude: 50
+    longitude: 9
+    ip_address: 10.50.9.1
+```
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: remove this

File: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md [hunk 19]
```diff
@@ -427,6 +429,167 @@ wan_virtual_topologies:
...
+
+AVD supports up to three tunnels (primary, secondary, tertiary).
+For each `zscaler` type Internet-policies, AVD uses the `cv_pathinfder_internet_exit_policies[name=<POLICY-NAME>].zscaler` dictionary and the `zscaler_endpoints` in combination with the `l3_interfaces.cv_pathfinder_internet_exit.policies[name=<POLICY-NAME>].tunnel_interface_numbers` to generate the internet-exit configuration.
+
+The `cv_pathinfder_internet_exit_policies[name=<POLICY-NAME>].zscaler` dictionary has additonnal options to configure the policy parameters shared with Zscaler through Cloudvision.
+
+!!! tip "IPsec"
+
+    When `true` the traffic going over the tunnels will be encrypted with AES-256-GCM.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md]: encrypt_..

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/zscaler_endpoints.schema.yml [hunk 20]
```diff
@@ -0,0 +1,68 @@
...
+# that can be found in the LICENSE file.
+# yaml-language-server: $schema=../../../../plugins/plugin_utils/schema/avd_meta_schema.json
+# Line above is used by RedHat's YAML Schema vscode extension
+# Use Ctrl + Space to get suggestions for every field. Autocomplete will pop up after typing 2 letters.
+type: dict
+keys:
+  zscaler_endpoints:
+    type: dict
+    documentation_options:
+      table: wan-internet-exit-settings
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/zscaler_endpoints.schema.yml]: remove this - make a this in its own table for Zscaler with a info about the loolup plugin

File: ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md [hunk 21]
```diff
@@ -798,6 +798,16 @@ roles/eos_designs/docs/tables/wan-virtual-topologies.md
...
+--8<--
+
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/docs/input-variables.md:810]: add subsection for zscaler

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_virtual_topology.schema.yml [hunk 22]
```diff
@@ -61,3 +61,11 @@ $defs:
                 "alternate" is converted to priority 2.
 
                 If not set, each path-group in `names` will be attributed its `default_preference`.
+      internet_exit:
+        type: dict
+        keys:
+          policy:
+            type: str
+            description: |-
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_virtual_topology.schema.yml:69]: make it preview

File: ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_virtual_topology.schema.yml [hunk 23]
```diff
@@ -61,3 +61,11 @@ $defs:
...
+              Internet-exit policy name associated with this virtual_topology.
+              The policy must be defined under `internet_exit_policies`.
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/schemas/schema_fragments/defs_virtual_topology.schema.yml]: wrong top level key

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 24]
```diff
@@ -585,3 +589,150 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+        # Only supported for CV Pathfinder
+        if not self.shared_utils.is_cv_pathfinder_router:
+            return []
+
+        internet_exit_policy_names = set()
+        candidate_internet_exit_policies = []
+        configured_internet_exit_policies = get(self._hostvars, "cv_pathfinder_internet_exit_policies", [])
+
+        # Look for internet-exit policy names
+        # TODO maybe a sub function for clarity
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 25]
```diff
@@ -585,3 +589,150 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+        for policy in self._filtered_wan_policies:
+            for match in get(policy, "matches", default=[]):
+                internet_exit_policy_name = match.get("internet_exit_policy_name")
+                if not internet_exit_policy_name or internet_exit_policy_name in internet_exit_policy_names:
+                    continue
+                # Maybe could use append_if_no_duplicate instead
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 26]
```diff
@@ -585,3 +589,150 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+            internet_exit_policy["connections"] = self.get_internet_exit_connections(internet_exit_policy)
+            if not internet_exit_policy["connections"]:
+                # No local interface for this policy
+                # TODO: Decide if we should raise here instead
+                continue
+            internet_exit_policies.append(internet_exit_policy)
+
+        return internet_exit_policies
+
+    def get_internet_exit_connections(self, internet_exit_policy) -> list:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py]:
```suggestion
    def get_internet_exit_connections(self, internet_exit_policy: dict) -> list:
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 27]
```diff
@@ -585,3 +589,150 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+          - Loop over wan_interfaces set on the policy
+            - For zscaler extract info from zscaler_endpoints
+            - Build connection dict with:
+              - source interface
+              - id? (something we can use to generate tunnel interface number)
+              - peer ip
+              - peer description
+              - type: <interface | tunnel>
+              - group: <policy_name>_PRI, _SEC, _TER
+              - ...
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py]:
```suggestion
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_internet_exit.py [hunk 28]
```diff
@@ -0,0 +1,59 @@
...
+        if not self.shared_utils.is_cv_pathfinder_router:
+            return None
+
+        router_internet_exit = {}
+        exit_groups_dict = {}
+        policies = []
+
+        for policy in self._filtered_internet_exit_policies:
+            policy_exit_groups = []
+            for connection in policy.get("connections", []):
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_internet_exit.py:37]: Add TODO as right now we rely on the order in connections. Need to make sure the order inside the policy is the expected one

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_adaptive_virtual_topology.py [hunk 29]
```diff
@@ -109,18 +111,33 @@ def _cv_pathfinder_profiles(self) -> list:
...
+
                 append_if_not_duplicate(
                     list_of_dicts=profiles,
                     primary_key="name",
-                    new_dict={"name": match["avt_profile"], "load_balance_policy": match["load_balance_policy"]["name"]},
+                    new_dict=profile,
                     context="Router Adaptive Virtual Topology profiles.",
                     context_keys=["name"],
                 )
             if (default_match := policy.get("default_match")) is not None:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_adaptive_virtual_topology.py:130]: Refactor - check if the default-match can be ignored in another PR

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/monitor_connectivity.py [hunk 30]
```diff
@@ -0,0 +1,69 @@
...
+        Only used for CV Pathfinder routers today
+        """
+        if not self.shared_utils.is_cv_pathfinder_router:
+            return None
+
+        monitor_connectivity = {}
+        interface_sets = []
+        hosts = []
+
+        # TODO mayb append if no duplicate
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/monitor_connectivity.py]:
```suggestion
```

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by gmuloc — COMMENTED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py [hunk 31]
```diff
@@ -0,0 +1,99 @@
...
+    async def set_swg_device(
+        self: CVClient,
+        device_id: str,
+        service: Literal["zscaler"],
+        location: str,
+        time: datetime | None = None,
+        timeout: float = 10.0,
+    ) -> EndpointConfig:
+        """
+        Get SWG Endpoints using arista.swg.v1.EndpointStatusService.GetOne API
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py:40]: s/Get SWG/Set SWG

s/GetOne/Set

File: ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py [hunk 32]
```diff
@@ -0,0 +1,99 @@
...
+    """
+
+    swg_api_version: Literal["v1"] = "v1"
+
+    async def set_swg_device(
+        self: CVClient,
+        device_id: str,
+        service: Literal["zscaler"],
+        location: str,
+        time: datetime | None = None,
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py:36]: There is no time parameter in set apis

File: ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py [hunk 33]
```diff
@@ -0,0 +1,99 @@
...
+        Returns:
+            EndpointStatus object matching the device_id
+        """
+        request = EndpointStatusStreamRequest(
+            partial_eq_filter=[
+                EndpointStatus(
+                    key=SwgKey(device_id=device_id, service=service),
+                ),
+            ],
+            time=time,
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/cv_client/client/swg.py:89]: this also seems incorrect, AFAIK, this field is supposed to be timebounds, not time.

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py [hunk 34]
```diff
@@ -236,3 +238,39 @@ def isis_decrypt(password: str, key: str, mode: str) -> str:
...
+    """
+    Encrypt (obfuscate) a password with insecure type-7.
+
+    Args:
+        data: The clear text password
+        salt: Optionally a number within the range 0-15. If not set, a random salt will be used.
+
+    Returns the encrypted password as a string.
+    """
+    if salt is None:
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:273]: IF salt is outside 0-15 throw error ?or auto-generate it ?

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py [hunk 35]
```diff
@@ -236,3 +238,39 @@ def isis_decrypt(password: str, key: str, mode: str) -> str:
...
+    Decrypt (deobfuscate) a password from insecure type-7.
+
+    Args:
+        data: The encrypted password
+
+    Returns the decrypted password as a string.
+    """
+    salt = int(data[0:2])
+    secret = bytearray.fromhex(data[2:])
+    return bytes(char ^ (SIMPLE_7_SEED[salt + i % 53]) for i, char in enumerate(secret)).decode("UTF-8")
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:260]: (salt + i ) % 53 instead of salt + (i % 53)

File: ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py [hunk 36]
```diff
@@ -236,3 +238,39 @@ def isis_decrypt(password: str, key: str, mode: str) -> str:
...
+    Args:
+        data: The clear text password
+        salt: Optionally a number within the range 0-15. If not set, a random salt will be used.
+
+    Returns the encrypted password as a string.
+    """
+    if salt is None:
+        salt = random.randint(0, 15)
+    cleartext = data.encode("UTF-8")
+    return f"{salt:02}" + bytearray(char ^ (SIMPLE_7_SEED[salt + i % 53]) for i, char in enumerate(cleartext)).hex().upper()
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/plugins/plugin_utils/password_utils/password.py:276]: ( salt + i ) % 53 instead of salt + i  % 53

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 37]
```diff
@@ -585,3 +589,133 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+            raise AristaAvdError(
+                f"Unsupported type '{internet_exit_policy['type']}' found in cv_pathfinder_internet_exit[name={internet_exit_policy['name']}]."
+            )
+
+        cloud_name = get(internet_exit_policy, "zscaler.cloud_name", required=True)
+        connections = []
+
+        # Check if the policy has any local interface
+        for wan_interface in self.shared_utils.wan_interfaces:
+            wan_interface_internet_exit_policies = get(wan_interface, "cv_pathfinder_internet_exit.policies", default=[])
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py:678]: Probably can belong to a shared util, which converts the wan_interface -> policy names 
to policy -> wan_interfaces map.

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py [hunk 38]
```diff
@@ -585,3 +589,133 @@ def _local_path_groups_connected_to_pathfinder(self) -> list:
...
+                "type": "tunnel",
+                "source_interface": wan_interface["name"],
+                "monitor_url": f"http://gateway.{cloud_name}.net/vpntest",
+                "ipsec_profile": "ZSCALER-IPSEC-PROFILE",
+            }
+
+            tunnel_id_range = range_expand(get(interface_policy_config, "tunnel_interface_numbers", required=True))
+
+            zscaler_endpoint_keys = ("primary", "secondary", "tertiary")
+            for index in range(len(zscaler_endpoint_keys)):
```
Comment (ayushmittal-arista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/utils.py:692]: for index, zscaler_endpoint_key in enumerate(zscaler_endpoint_keys)
